### PR TITLE
FEAT(client): Adds shortcut to listen to a channel

### LINKED ISSUE #4486: Shortcut to listen to a channel
I believe that this implementation can be used for broadcast communication and also for games that want to hear a channel that is rolling a song for example.

Could you put a shortcut so you can start and end listening to a certain channel?
If you can have the implementation of a shortcut also to adjust the level of the channel. It will be perfect.

### PR DESCRIPTION
**Pull request by timjwkim**
Added a global shortcut that toggles listening to a specific channel targeted by the client. 

Fixes #4486

### CODE REVIEW

File: src/mumble/MainWindow.cpp [hunk 1]
```diff
@@ -225,6 +225,10 @@ void MainWindow::createActions() {
 	gsJoinChannel->setObjectName(QLatin1String("MetaChannel"));
 	gsJoinChannel->qsToolTip = tr("Use in conjunction with Whisper to.", "Global Shortcut");
 
+	gsListenChannel = new GlobalShortcut(this, idx++, tr("Listen Channel", "Global Shortcut"));
```
Comment (Krzmbrzl) [inline src/mumble/MainWindow.cpp:228]:
```suggestion
	gsListenChannel = new GlobalShortcut(this, idx++, tr("Listen to Channel", "Global Shortcut"));
```
Or is that too long to fit in the UI? :thinking:

File: src/mumble/MainWindow.cpp [hunk 2]
```diff
@@ -3040,6 +3044,21 @@ void MainWindow::on_gsToggleMainWindowVisibility_triggered(bool down, QVariant)
 	}
 }
 
+void MainWindow::on_gsListenChannel_triggered(bool down, QVariant scdata) {
+	ShortcutTarget st = scdata.value< ShortcutTarget >();
+	Channel *c = mapChannel(st.iChannel);
```
Comment (Krzmbrzl) [inline src/mumble/MainWindow.cpp:3049]:
```suggestion
	const Channel *c = mapChannel(st.iChannel);
```

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]
Please ensure that your commit message follows the format specified in our [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md)

**Comment by Krzmbrzl** [comment]
@timjwkim are you going to update this PR or shall I finish it up for you?

**Comment by timjwkim** [comment]
My apologies, didn't get a chance to see the updates on this thread til now.
I will finish up the PR!

**Comment by Krzmbrzl** [comment]
@timjwkim so?

**Comment by kol0** [comment]
@Krzmbrzl What can I do to push this change forward?
I would love to be able to set up channel listening with hotkeys.

**Comment by Krzmbrzl** [comment]
@kol0 as it stands I think it would be fine, if you forked @timjwkim repo and pick up the work from where it was left. As long as you don't remove @timjwkim from the commit authorship (but add yourself), I think that should be fine.

**Review by Krzmbrzl — APPROVED** [review]
I took the liberty of finishing this PR myself. In the end it did turn out to be a bit more complicated than originally anticipated due to the way server-specfic global shortcuts were handled up to know.

**Pull request merged**